fix: display:none should be considered by only-listitems and only-dlitems

erikkroes opened the pull request:
- Added a condition to only-listitems and only-dlitems to consider if an element is set to display:none.

Closes issue:

- https://github.com/dequelabs/axe-core/issues/1021

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [ ] Follows the commit message policy, appropriate for next version
- [ ] Has documentation updated, a DU ticket, or requires no documentation change
- [ ] Includes new tests, or was unnecessary
- [ ] Code is reviewed for security by: << Name here >>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dequelabs/axe-core?pullRequest=1022) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dequelabs/axe-core?pullRequest=1022) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dequelabs/axe-core?pullRequest=1022) it.</sub>

jkodu commented on lib/checks/lists/only-listitems.js:
```diff
@@ -17,6 +17,13 @@ const getHasListItem = (hasListItem, tagName, isListItemRole) => {
 
+const getIsHidden = actualNode => {
+	return (
+		window.getComputedStyle(actualNode, null).getPropertyValue('display') ===
```
DRY, worth extracting this method to axe.utils

erikkroes replied:
I wasn't aware of the utils, and I see it actually has an is-hidden.js already. Guess I can rewrite using that.

jkodu replied:
Because of the above disparity, I have put in this PR - https://github.com/dequelabs/axe-core/pull/1023 which will give an idea of API methods available.

WilcoFiers replied:
I don't think this should use the existing isHidden method, as that looks up the tree. We don't need that here. Just check for `display:none;` and `visibility:hidden;`.

erikkroes replied:
Still worth extracting then?

marcysutton replied:
I think so, yes. See my comment above about `aria-hidden="true"`

erikkroes commented on lib/checks/lists/only-dlitems.js:
```diff
@@ -11,19 +11,32 @@ const ALLOWED_TAGS = [
+
 		if ((tagName !== 'DT' && tagName !== 'DD') || role) {
-			if (!ALLOWED_ROLES.includes(role)) {
+			if (!ALLOWED_ROLES.includes(role) && !isHidden) {
```
I followed the style of only-listitems.js. I found it to be more readable, but can change it if you like. The two files are already quite different from each other. More than I expected them to be.

jkodu commented on lib/checks/lists/only-dlitems.js:
```diff
@@ -11,19 +11,32 @@ const ALLOWED_TAGS = [
 
+const getIsHidden = actualNode => {
+	return (
+		window.getComputedStyle(actualNode, null).getPropertyValue('display') ===
```
@WilcoFiers - your view on above is much appreciated.

marcysutton replied:
The only styles that should apply to this are ones that impact the accessibility tree (I use the [Chrome A11y Devtools inspector](https://bit.ly/chrome-a11y) for this). Erik is right that opacity:0 is still exposed, so without some other CSS style affecting its render state it shouldn't fail by itself. 

Visibility: visible on a child is interesting, I didn't realize that could be overridden. `aria-hidden="true"` doesn't work the same way–it applies to whole HTML subtrees and can't be overridden. `aria-hidden="true"` would be appropriate to cover here.

erikkroes commented:
The 1001 file changes come from some file permissions that I changed, and came along. Fixed that locally already. Will pick up the rest of the comments as well.

WilcoFiers commented:
Superseded by #1022. Want to get it into the release of Monday, so I took it over as there didn't seem to be any progress on it anymore.

team-a11y commented:
Ah sorry for keeping you waiting. Other things got in the way. Great to see it got picked up!

erikkroes commented:
Ah sorry for keeping you waiting. Other things got in the way. Great to see it got picked up!
